larray/core/session.py

```diff
@@ -820,44 +821,56 @@ def array_equals(self, other):
 
         Examples
         --------
-        >>> s1 = Session([('arr1', ndtest(2)), ('arr2', ndtest((2, 2)))])
-        >>> s2 = Session([('arr1', ndtest(2)), ('arr2', ndtest((2, 2)))])
-        >>> s1.array_equals(s2)
-        name  arr1  arr2
-              True  True
+        >>> a = Axis('a=a0..a2')
+        >>> a01 = a['a0,a1'] >> 'a01'
+        >>> s1 = Session([('a', a), ('a01', a01), ('arr1', ndtest(2)), ('arr2', ndtest((2, 2)))])
+        >>> s2 = Session([('a', a), ('a01', a01), ('arr1', ndtest(2)), ('arr2', ndtest((2, 2)))])
+        >>> s1.element_equals(s2)
+        name     a   a01  arr1  arr2
+              True  True  True  True
 
-        Different value(s)
+        Different value(s) between two arrays
 
         >>> s2.arr1['a1'] = 0
-        >>> s1.array_equals(s2)
-        name   arr1  arr2
-              False  True
+        >>> s1.element_equals(s2)
+        name     a   a01   arr1  arr2
+              True  True  False  True
 
         Different label(s)
 
         >>> s2.arr2 = ndtest("b=b0,b1; a=a0,a1")
-        >>> s1.array_equals(s2)
-        name   arr1   arr2
-              False  False
+        >>> s2.a = Axis('a=a0,a1')
+        >>> s1.element_equals(s2)
+        name      a   a01   arr1   arr2
+              False  True  False  False
 
-        Extra/missing array(s)
+        Extra/missing axis(es), group(s), array(s)
 
         >>> s2.arr3 = ndtest((3, 3))
-        >>> s1.array_equals(s2)
-        name   arr1   arr2   arr3
-              False  False  False
+        >>> del s2.a
+        >>> s1.element_equals(s2)
+        name      a   a01   arr1   arr2   arr3
+              False  True  False  False  False
         """
-        self_keys = set(self.keys())
-        all_keys = list(self.keys()) + [n for n in other.keys() if n not in self_keys]
-        def larray_nan_equal(a1, a2):
-            try:
-                a1 = aslarray(a1)
-            except Exception:
+        self_keys =[k for k, v in self.items() if isinstance(v, (Axis, Group, LArray))]
+        other_keys = [k for k, v in other.items() if isinstance(v, (Axis, Group, LArray)) and k not in self_keys]
+        all_keys = self_keys + other_keys
+
+        def elem_equal(e1, e2):
+            if not type(e1) is type(e2):
                 return False
-            return a1.equals(a2, nan_equals=True)
-        res = [larray_nan_equal(self.get(key), other.get(key)) for key in all_keys]
+            if isinstance(e1, Group):
+                return e1 == e2
```

Comment:
This is wrong if the group keys are arrays:

```python
>>> a = Axis('a=a0..a2')
>>> a['a0,a2'] == a['a1,a2']
False
>>> a[np.array(['a0', 'a2'])] == a[np.array(['a1', 'a2'])]
array([False,  True], dtype=bool)
```
and besides, I wonder if we should not go beyond == for groups, because == does not test .name. 

Maybe we should implement Group.equals which would solve both problems at the same time.

Reply:
Seems a good idea. Do I open a new issue for that?

Reply:
do as you prefer.

Reply:
I will add a changelog for Group.equals but no new issue.